testutils/mockhyperbahn/hyperbahn.go

```diff
@@ -108,11 +123,23 @@ func (h *Mock) Configuration() hyperbahn.Configuration {
 	}
 }
 
+// Channel returns the underlying tchannel that implements relaying.
+func (h *Mock) Channel() *tchannel.Channel {
+	return h.ch
+}
+
+// getSC returns the isolated subchannel for the service.
+func (h *Mock) getSC(svc string) *tchannel.SubChannel {
+	return h.ch.GetSubChannel(svc, tchannel.Isolated)
+}
+
 func (h *Mock) adHandler(ctx json.Context, req *hyperbahn.AdRequest) (*hyperbahn.AdResponse, error) {
+	callerHostPort := tchannel.CurrentCall(ctx).RemotePeer().HostPort
 	h.Lock()
 	for _, s := range req.Services {
 		h.advertised = append(h.advertised, s.Name)
-		h.table.addPeer(s.Name, tchannel.CurrentCall(ctx).RemotePeer().HostPort)
+		sc := h.ch.GetSubChannel(s.Name, tchannel.Isolated)
```

Comment:
Removed `h.getSC`, I wrote it thinking we'd use it in multiple places, but the table is a separate type.